cmd/kube-proxy/app/server.go

```diff
@@ -914,30 +915,75 @@ func (s *ProxyServer) birthCry() {
 //
 // The order of precedence is:
 //  1. if bindAddress is not 0.0.0.0 or ::, then it is used as the primary IP.
-//  2. if the Node object can be fetched, then its primary IP is used as the primary IP
-//     (and its secondary IP, if any, is just ignored).
-//  3. otherwise the primary node IP is 127.0.0.1.
+//  2. if the Node object can be fetched, then its address(es) is/are used
+//  3. otherwise the node IPs are 127.0.0.1 and ::1
 //
 // In all cases, the secondary IP is the zero IP of the other IP family.
 func detectNodeIPs(client clientset.Interface, hostname, bindAddress string) (v1.IPFamily, map[v1.IPFamily]net.IP) {
-	nodeIP := netutils.ParseIPSloppy(bindAddress)
-	if nodeIP.IsUnspecified() {
-		nodeIP = utilnode.GetNodeIP(client, hostname)
+	primaryFamily := v1.IPv4Protocol
+	nodeIPs := map[v1.IPFamily]net.IP{
```

Comment:
is not better to structure the code same as the comment?

```
//  1. if bindAddress is not 0.0.0.0 or ::, then it is used as the primary IP.
//  2. if the Node object can be fetched, then its address(es) is/are used
//  3. otherwise the node IPs are 127.0.0.1 and ::1
```

that way is straightforward to read

Reply:
I'm ok with the current organization...